arngll/src/frame_info.rs

```diff
@@ -0,0 +1,648 @@
+// Copyright (c) 2022, The ARNGLL-Rust Authors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+use std::fmt::{Debug, Formatter};
+use super::*;
+
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub struct NetworkId(pub u16);
+
+impl NetworkId {
+    pub fn from_iter<'a, T: Iterator<Item=&'a u8>>(iter: &mut T) -> NetworkId {
+        let msb = *iter.next().unwrap();
+        let lsb = *iter.next().unwrap();
+        NetworkId(((msb as u16)<<8) + (lsb as u16))
```

Comment:
I'm slightly surprised to see addition instead of a bitwise or here, but logically it makes no difference.

Reply:
I'll go ahead and make it OR